File: Python/剑指 Offer 57 - II. 和为s的连续正数序列.py
class Solution:
    def findContinuousSequence(self, target: int) -> List[List[int]]:
        l=1
        r=1

        sum1=0
        ans=[]
        while r<=(target+1)//2+1:
            sum1+=r
            while sum1>target:
                sum1-=l
                l+=1
            if sum1==target:
                if r-l+1>=2:
                    ans.append(list(range(l,r+1)))
            
            r+=1
        return ans
            

File: Reigo/leetcode/JavaScript/剑指 Offer 57 - II. 和为s的连续正数序列.js
/**
 * @param {number} target
 * @return {number[][]}
 */
 var findContinuousSequence = function(target) {
    if(target==1){
        return [];
    }
    var l=1;
    var r=2;
    var sum=1;
    var ans=[];
    while(l<=parseInt(target/2)){
        
        while(sum<target){
            sum+=r;
            r+=1;
        }
        if(sum==target){
            var temp=[];
            for(var i=l;i<r;i++){
                temp.push(i);
            }
            ans.push(temp);
        }
        sum=sum-l;
        l+=1;
    }
    return ans;
};